metrics/timescaledb/db.py

```diff
@@ -143,7 +144,40 @@ def _drop_child_tables(connection, table):
 
 
 def _ensure_table(connection, table):
-    connection.execute(schema.CreateTable(table, if_not_exists=True))
+    def alter_table(table, engine):
+        return f"ALTER TABLE {table.name}"
+
+    def add_column(table, col, engine):
+        col_type = col.type.compile(dialect=engine.dialect)
+
+        if col.nullable:
+            nullability = "NULL"
+        else:
+            if not col.default:
+                raise ValueError("Adding a nullable column requires a default value")
+            type_processor = col.type.literal_processor(dialect=engine.dialect)
+            processed_default = type_processor(col.default.arg)
+            nullability = f"NOT NULL DEFAULT {processed_default}"
+        return f"{alter_table(table, engine)} ADD {col.name} {col_type} {nullability}"
+
+    def drop_column(table, col, engine):
+        return f"{alter_table(table, engine)} DROP COLUMN {col.name}"
+
+    metadata = MetaData()
+    engine = connection.engine
+    metadata.reflect(engine)
+    if table.name not in metadata.tables:
+        connection.execute(schema.CreateTable(table))
+    else:
+        db_table = metadata.tables[table.name]
+        for column in [
+            c for c in table.columns if c.name not in [d.name for d in db_table.columns]
+        ]:
+            connection.execute(text(add_column(table, column, engine)))
+        for column in [
+            d for d in db_table.columns if d.name not in [c.name for c in table.columns]
+        ]:
+            connection.execute(text(drop_column(table, column, engine)))
```

Comment:
> I don't think we should make alter_table, add_column, and drop_column inner functions of _ensure_table. We don't need the closure; or we don't make use of it, because we pass table and engine, which we obtain from connection, to all three. So, if we are going to make them inner functions, then we should make use of the closure.

I agree. The query composition logic was previously a bit more complex but I found I could radically simplify it and thus those functions are pretty redundant. I'll remove them.

>I think the listcomp in a listcomp in a for loop hard to parse. Could we obtain columns to add and columns to drop, and then operate on them?

Yeah, I agree that'd be a lot clearer